src/main/java/com/dynatrace/jenkins/dashboard/steps/TATestRunRegistrationPipelineStep.java

```diff
@@ -0,0 +1,112 @@
+package com.dynatrace.jenkins.dashboard.steps;
+
+import com.dynatrace.jenkins.dashboard.Messages;
+import com.dynatrace.jenkins.dashboard.TATestRunRegistrationBuildStep;
+import com.dynatrace.sdk.server.testautomation.models.TestCategory;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.util.ListBoxModel;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.workflow.steps.*;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+/**Extracted to new step, because we need to return a value (testRunId) from step*/
```

Comment:
explanations like that aren't javadocs, but simple (single- or multiline) comments